ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java

```diff
@@ -15,48 +15,45 @@
  * limitations under the License.
  */
 
-package org.jboss.as.quickstarts.ha.singleton.service.primary;
+package org.jboss.as.quickstarts.ha.singleton.service;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
 import org.jboss.logging.Logger;
-import org.jboss.msc.service.Service;
-import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.Service;
 import org.jboss.msc.service.StartContext;
-import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
 import org.wildfly.clustering.group.Node;
-
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import org.wildfly.clustering.service.ActiveServiceSupplier;
 
 /**
- * This service will regularly query the singleton service and log where the singleton service is running. For the sake
- * of illustration, the value returned by the singleton service {@link SingletonService#getValue()} is the node where it
- * is currently running.
+ * This service periodically looks up the singleton service. If the service is running on this node it will log value provided
+ * by the service. For the sake of illustration, singleton service supplies a value which represents the node where it is
+ * currently running.
  *
  * @author Radoslav Husar
  */
-class QueryingService implements Service<Void> {
+class QueryingService implements Service {
 
     private Logger LOG = Logger.getLogger(this.getClass());
     private ScheduledExecutorService executor;
 
     @Override
-    public void start(StartContext context) throws StartException {
+    public void start(StartContext context) {
         LOG.info("Querying service is starting.");
 
         executor = Executors.newSingleThreadScheduledExecutor();
         executor.scheduleAtFixedRate(() -> {
 
-            @SuppressWarnings("unchecked")
-            ServiceController<Node> service = (ServiceController<Node>) context.getController().getServiceContainer()
-                    .getService(ServiceActivator.SINGLETON_SERVICE_NAME);
-            try {
-                Node node = service.awaitValue(5, TimeUnit.SECONDS);
-                LOG.infof("Singleton service is running on node '%s'.", node);
-            } catch (InterruptedException | TimeoutException | IllegalStateException e) {
-                LOG.warn("Failed to query singleton service.");
+            Supplier<Node> node = new ActiveServiceSupplier<>(context.getController().getServiceContainer(), SingletonServiceActivator.SINGLETON_SERVICE_NAME);
```

Comment:
This should use PassiveServiceSupplier, not ActiveServiceSupplier.

Reply:
Done